framework/src/variables/MooseVariableData.C

```diff
@@ -1330,6 +1330,16 @@ MooseVariableData<OutputType>::computeAD(const unsigned int num_dofs, const unsi
   mooseAssert(_var.kind() == Moose::VarKindType::VAR_AUXILIARY || ad_offset || !_var_num,
               "Either this is the zeroth variable or we should have an offset");
 
+#ifndef MOOSE_SPARSE_AD
+  if (ad_offset + num_dofs > MOOSE_AD_MAX_DOFS_PER_ELEM)
+    mooseError("Current number of dofs per element ",
+               ad_offset + num_dofs,
```

Comment:
This is really helpful, so I know exactly what to change it too.

Reply:
So when this error kicks out at run time, it isn't always the full dofs. For example, a problem I run needs 102, which I only found out by this series of errors
```
Current number of dofs per element 54 is greater than AD_MAX_DOFS_PER_ELEM of 50. You can run `configure --with-derivative-size=<n>` to request a larger derivative container.
```
```
Current number of dofs per element 63 is greater than AD_MAX_DOFS_PER_ELEM of 54. You can run `configure --with-derivative-size=<n>` to request a larger derivative container.
```
```
Current number of dofs per element 102 is greater than AD_MAX_DOFS_PER_ELEM of 81. You can run `configure --with-derivative-size=<n>` to request a larger derivative container.
```
Is there a way to do an accurate accounting before spitting out the error? If not, it may be worthwhile to print out the DOFs for a successful run (here 102) so the user can optimize `MOOSE_AD_MAX_DOFS_PER_ELEM`

Reply:
i.e., print out in the header info, maybe here?
```
Nonlinear System:
  Num DOFs:                35735
  Num Local DOFs:          8975
  Variables:               { "disp_x" "disp_y" "temp" } "lm" 
  Finite Element Types:    "LAGRANGE" "LAGRANGE" 
  Approximation Orders:    "SECOND" "SECOND" 
```

Reply:
I can probably print it fairly close to there